main.go

```diff
@@ -0,0 +1,475 @@
+	burst                 int     = 1
+	maxPageSize           int64   = 1000
+	assetTypeOrganization string  = "cloudresourcemanager.googleapis.com/Organization"
+	assetTypeProject      string  = "cloudresourcemanager.googleapis.com/Project"
+)
+
+type userAccess struct {
+	User          admin.User
+	Organizations []*cloudresourcemanager.Organization
+	Projects      []*cloudresourcemanager.Project
+	Error         error
+}
+
+func main() {
+
+	var wg sync.WaitGroup
+	serviceAccountFile := flag.String("serviceAccountFile", "", "Service Account JSON files with IAM permissions to the org")
+	impersonatedServiceAccount := flag.String("impersonatedServiceAccount", "", "Impersonated Service Accounts the script should run as")
+	organization := flag.String("organization", "", "The organizationID that is the subject of this audit")
+	subject := flag.String("subject", "", "The admin user to for the organization that can use the Directory API to list users")
+	cx := flag.String("cx", "", "Workspace Customer ID number")
+	delay := flag.Int("delay", 1*1000, "delay in ms for each user iterated")
+
+	flag.Parse()
+	defer glog.Flush()
+	ctx := context.Background()
+
+	// Configure a rate limiter that will control how frequently API calls to
+	// Cloud Resource Manager is made.
+	limiter := rate.NewLimiter(rate.Limit(maxRequestsPerSecond), burst)
+
+	// Initialize a randomSeed for use later
+	rand.Seed(time.Now().UnixNano())
+
+	if *organization == "" || *subject == "" || *cx == "" {
+		glog.Error("--organization, --cx and --subject must be specified")
+		return
+	}
+	if (*serviceAccountFile == "" && *impersonatedServiceAccount == "") || (*serviceAccountFile != "" && *impersonatedServiceAccount != "") {
+		glog.Error("either --serviceAccountFile or --impersonatedServiceAccount must be specified")
+		return
+	}
+
+	// Parse the serviceAccount names or keys provided. Multiple values can be set to shard API quota between projects.
+	serviceAccounts, svcAccountKeys, err := parseServiceAccounts(*impersonatedServiceAccount, *serviceAccountFile)
+	if err != nil {
+		glog.Errorf("Error parsing serviceAccounts %v", err)
+		return
+	}
+
+	// Select an random serviceAccount to use.
+	impersonateAccount, svcKeyCred := getRandomServiceAccount(serviceAccounts, svcAccountKeys)
+
+	// Initialize the Cloud Asset API.
+	// This api is used to find the projects in an ORG
+	assetClient, err := getAssetClient(ctx, impersonateAccount, svcKeyCred)
+	if err != nil {
+		glog.Errorf("Error getting Cloud Asset API Client %v", err)
+		return
+	}
+
+	// Use the Cloud Asset API to recall/find the Organization for provided organizationID.
+	// We do not really need to do this section since the the orgID was provided in the input argument.
+	// We are also setting a default "emptyQuery" value here to return all projects.
+	// We are doing this as a type of input validation with the asset-api to search/narrow the organization.
+	emptyQuery := ""
```

Comment:
Suggestion, feel free to ignore. `findResourcesByAssetType` only ever uses `emptyQuery`. Consider either documenting possible query values or removing that parameter entirely.